src/theory/datatypes/datatypes_sygus.cpp

```diff
@@ -1127,11 +1149,22 @@ Node SygusSymBreakNew::registerSearchValue(
         Node bad_val_o = d_cache[a].d_search_val[tn][bad_val_bvr];
         Assert( d_cache[a].d_search_val_sz[tn].find( bad_val_bvr )!=d_cache[a].d_search_val_sz[tn].end() );
         unsigned prev_sz = d_cache[a].d_search_val_sz[tn][bad_val_bvr];
-        if( prev_sz>sz ){
+        bool doFlip = (prev_sz > sz);
+        if (doFlip)
+        {
           //swap : the excluded value is the previous
           d_cache[a].d_search_val_sz[tn][bad_val_bvr] = sz;
           bad_val = d_cache[a].d_search_val[tn][bad_val_bvr];
           bad_val_o = nv;
+          if (Trace.isOn("sygus-sb-exc"))
```

Comment:
you do not need to guard this with `Trace.isOn`. The call to `quantifiers::TermDbSygus::toStreamSygus` already does this test and the other trace instructions will as well.

Reply:
This still saves two calls and 2 checks.